api/internal/testutil/server.go

```diff
@@ -307,23 +296,30 @@ func (s *TestServer) waitForAPI() {
 // available, and then waits for a known leader and an index of
 // 1 or more to be observed to confirm leader election is done.
 func (s *TestServer) waitForLeader() {
-	WaitForResult(func() (bool, error) {
+	f := func() error {
 		// Query the API and check the status code
 		// Using this endpoint as it is does not have restricted access
 		resp, err := s.HTTPClient.Get(s.url("/v1/status/leader"))
 		if err != nil {
-			return false, err
+			return fmt.Errorf("failed to get leader: %w", err)
 		}
-		defer resp.Body.Close()
-		if err := s.requireOK(resp); err != nil {
-			return false, err
+		defer func() { _ = resp.Body.Close() }()
+		if err = s.requireOK(resp); err != nil {
+			return fmt.Errorf("leader response is not ok: %w", err)
 		}
+		return nil
+	}
 
-		return true, nil
-	}, func(err error) {
-		defer s.Stop()
-		s.t.Fatalf("err: %s", err)
-	})
+	test.Wait(s.t,
+		wait.On(
+			wait.ErrorFunc(f),
```

Comment:
How wedded to the name `ErrorFunc` are you vs something like `UntilNilError` or `UntilNoError` (or whatever)? The name could just as easily be saying we're waiting for an error to be non-nil.

Reply:
Definitely open to changing the names; let me noodle on that while working on the inverse of wait.On, https://github.com/shoenig/test/issues/80